forge-gui/src/main/java/forge/gamemodes/match/input/InputPayMana.java

```diff
@@ -345,7 +346,12 @@ else if (colorMatches.size() < abilitiesMap.size()) {
         game.getAction().invoke(new Runnable() {
             @Override
             public void run() {
+                ManaConversionMatrix old = new ManaConversionMatrix();
```

Comment:
not ideal
we might just have to do a full recalculation for new boardstate
the only problem is carrying over playerManaConversion? 🤔

Reply:
I think the spent mana stuff might be locked-in when starting to cast/activate abilities

Otherwise I'm going to ask Judges

Reply:
so far this example:

* Vizier of the Menagerie on the field, 
* cast Bayou Groff, sacrifice Vizier as additional Cost

Now the problem, technically you should be able to pay the costs in any order, but Forge doesn't support that

then IF you sacrifice Vizier before paying the Mana, then the apply would/should be lost

Reply:
I guess then we can worry more about it when #309 has progress

Reply:
yeah, i think it might be better to allow that for now